src/Grid.php

```diff
@@ -504,13 +504,14 @@ public function addPopup($columnName, $popup = null, $icon = 'caret square down'
      * @param string|array|View                 $button
      * @param string                            $title
      * @param \Closure(View, string|null): void $callback
-     * @param array                             $args     extra URL argument for callback
+     * @param array                             $args       extra URL argument for callback
+ 	* @param \Closure|bool                      $isDisabled function (Model $row) { ... }
      *
      * @return View
      */
-    public function addModalAction($button, $title, \Closure $callback, $args = [])
+    public function addModalAction($button, $title, \Closure $callback, $args = [], $isDisabled = false)
```

Comment:
bulk method was added - https://github.com/atk4/ui/pull/1920 - adjust this PR to support both methods